src/test/java/com/snowplowanalytics/snowplow/tracker/payload/SelfDescribingJsonTest.java

```diff
@@ -67,4 +67,30 @@ public void testMakeSdjWithSdj() {
         assertNotNull(sdj);
         assertEquals(expected, sdjString);
     }
+
+    @Test
+    public void testEqualityOfTwoInstances_withSchemaNameOnly() {
+        SelfDescribingJson a = new SelfDescribingJson("schema");
+        SelfDescribingJson b = new SelfDescribingJson("schema");
+        assertEquals(a, b);
+    }
+
+    @Test
+    public void testEqualityOfTwoInstances_withTrackerPayload() {
+        TrackerPayload nestedData = new TrackerPayload();
+        nestedData.add("key", "value");
+        SelfDescribingJson a = new SelfDescribingJson("schema", nestedData);
+        SelfDescribingJson b = new SelfDescribingJson("schema", nestedData);
+        assertEquals(a, b);
+    }
+
+    @Test
+    public void testEqualityOfTwoInstances_withNestedEvent() {
+        TrackerPayload nestedData = new TrackerPayload();
+        nestedData.add("key", "value");
+        SelfDescribingJson nestedEvent = new SelfDescribingJson("nested_event", nestedData);
+        SelfDescribingJson a = new SelfDescribingJson("schema", nestedEvent);
+        SelfDescribingJson b = new SelfDescribingJson("schema", nestedEvent);
+        assertEquals(a, b);
+    }
```

Comment:
This is just my paranoia, but can you please add these extra tests so that we also test for non-equality:

```java

    @Test
    public void testEqualityOfTwoInstances_withTwoMapInstances() {
        Map<String, Object> data1 = new HashMap<>();
        data1.put("key", "value");
        SelfDescribingJson a = new SelfDescribingJson("schema", data1);

        Map<String, Object> data2 = new HashMap<>();
        data2.put("key", "value");
        SelfDescribingJson b = new SelfDescribingJson("schema", data2);

        assertEquals(a, b);
    }

    @Test
    public void testEqualityOfTwoInstances_withNotEqualInstances() {
        Map<String, Object> data1 = new HashMap<>();
        data1.put("key", "value1");
        SelfDescribingJson a = new SelfDescribingJson("schema1", data1);

        Map<String, Object> data2 = new HashMap<>();
        data2.put("key", "value2");
        SelfDescribingJson b = new SelfDescribingJson("schema1", data2);

        SelfDescribingJson c = new SelfDescribingJson("schema2", data2);

        assertNotEquals(a, b);
        assertNotEquals(b, c);
    }
```

Reply:
Of course I can, I will do them in a little bit :)